src/blob/core.rs

```diff
@@ -250,7 +261,7 @@ where
         let write_result = partially_serialized.write_to_file(&blob.file).await?;
         header.set_offset_checksum(write_result.blob_offset(), write_result.header_checksum());
         blob.index.push(key, header)?;
-        Ok(())
+        Ok(WriteResult { dirty_bytes: blob.file.dirty_bytes() })
```

Comment:
I think you can expose `dirty_bytes` as a function inside `Blob` structure. In this way there is no need to update result type of `write` function

Reply:
We acquire readlock here, and if we expose it as a function we'll need to acquire another readlock on write